config/routes.rb

```diff
@@ -43,14 +43,18 @@ def put(*_args); end
         post :copy, :action => 'copy', :controller => 'portfolios'
         post :undelete, :action => 'restore', :controller => 'portfolios'
       end
+      post '/portfolio_items/:portfolio_item_id/icon', :to => 'portfolio_items#add_icon_to_portfolio_item'
       resources :portfolio_items,       :only => [:create, :destroy, :index, :show, :update] do
         resources :provider_control_parameters, :only => [:index]
         resources :service_plans,               :only => [:index]
-        get :icon, :action => 'show', :controller => 'icons'
+        get :icon, :to => 'icons#show_icon'
```

Comment:
Wait, doesn't this not work? Because we're not passing in an icon id, we're passing in a `portfolio_item_id`, and thus when it hits the `icons_controller#show_icon` method, it's going to do nothing, right? I don't see a spec in `requests/portfolio_items_spec.rb` regarding icons, so maybe that's why there's nothing failing at the moment.

Reply:
Updated the method name to be `raw_icon` rather than `show_icon`. Bad naming on my part!